### PR TITLE
fix: snaps carousel back to first image when autoscrolling

### PR DESCRIPTION
**Pull request by winescout**
In this PR
=========
Amends carousel behavior to snap back to first item when autoscrolling and reaches the end
Adds Storybook to demo the behavior

https://www.screencast.com/t/GM6UfK168m

### CODE REVIEW

File: packages/fscomponents/src/components/MultiCarousel/MultiCarousel.web.tsx [hunk 1]
```diff
@@ -281,9 +281,13 @@ export const MultiCarousel = <ItemT, >(props: MultiCarouselProps<ItemT>) => {
   );
 
   const goToNext = useCallback(async (options?: GoToOptions | GestureResponderEvent) => {
-    const nextIndex = currentIndex + 1 > numberOfPages - 1 ? numberOfPages - 1 : currentIndex + 1;
+    const nextIndex = currentIndex + 1 > numberOfPages - 1 ? 0 : currentIndex + 1;
+    const animationOptions =
```
Comment (ckornell) [inline packages/fscomponents/src/components/MultiCarousel/MultiCarousel.web.tsx:285]: All my same comments from #2324 apply. 

https://github.com/brandingbrand/flagship/pull/2324

**Review by ckornell — CHANGES_REQUESTED** [review]

File: packages/fscomponents/src/components/MultiCarousel/MultiCarousel.web.tsx (hunk 1 above)
Comment (winescout) [reply packages/fscomponents/src/components/MultiCarousel/MultiCarousel.web.tsx]: Cleaned up this code with @ckornell , and removed the need for this line

**Comment by winescout** [comment]
Merging these two approaches into this PR:  https://github.com/brandingbrand/flagship/pull/2323

**Review by ckornell — APPROVED** [review]

**Review by wSedlacek — APPROVED** [review]
This looks good.

**Pull request merged**